invenio_communities/communities/permissions.py

```diff
@@ -1,106 +1,56 @@
 # -*- coding: utf-8 -*-
 #
 # This file is part of Invenio.
-# Copyright (C) 2020 CERN.
+# Copyright (C) 2020-2021 CERN.
 #
 # Invenio is free software; you can redistribute it and/or modify it
 # under the terms of the MIT License; see LICENSE file for more details.
 
 """Community permissions."""
 
 from flask_principal import UserNeed
-from flask_security import current_user
-from invenio_access.permissions import any_user
-from invenio_records_permissions.generators import AnyUser, Generator
+from invenio_records_permissions.generators import Generator, \
+    AuthenticatedUser
 from invenio_records_permissions.policies import BasePermissionPolicy
-
-# from invenio_communities.members.permissions import CommunityMember
+from invenio_access.permissions import any_user
+from elasticsearch_dsl.query import Q
 
 
-class CommunityVisibilityPolicy(Generator):
+class AnyUserIfPublic(Generator):
     """Allows only community members to read the community if its hidden."""
 
-    def needs(self, community=None, **kwargs):
-        """."""
-        if community['visibility'] == 'public':
-            return [any_user]
-        else:
-            return [CommunityMember(['admin', 'curator', 'member'])]
+    def needs(self, record=None, **kwargs):
+        """Visibility Needs."""
+        # visibility = (
+        #     record and
+        #     record.get('access', {}).get('visibility', 'open')
+        # )
+        #  if not visibility == 'open' else [UserNeed(record['metadata']['created_by'])]
+        return [any_user]
+
+    def query_filter(self, **kwargs):
+        """Filters for non-restricted records."""
+        # TODO: Implement with new permissions metadata
+        return Q('term', **{"_access.metadata_restricted": False})
```

Comment:
`_access` is not implemented yet right?

Reply:
not yet but since this is elastic search it will just default to ```False```

Reply:
The field is now `access.visibility: "public"`